dlpy/tests/test_network.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# NOTE: This test requires a running CAS server.  You must use an ~/.authinfo
+#       file to specify your username and password.  The CAS host and port must
+#       be specified using the CASHOST and CASPORT environment variables.
+#       A specific protocol ('cas', 'http', 'https', or 'auto') can be set using
+#       the CASPROTOCOL environment variable.
+#
+
+import swat
+import swat.utils.testing as tm
+from dlpy.network import Network
+from dlpy.model import Model
+from dlpy.layers import *
+from dlpy.utils import DLPyError
+
+
+class TestNetwork(tm.TestCase):
+    # Create a class attribute to hold the cas host type
+    server_type = None
+    s = None
+
+    @classmethod
+    def setUpClass(cls):
+        swat.reset_option()
+        swat.options.cas.print_messages = False
+        swat.options.interactive_mode = False
+
+        cls.s = swat.CAS()
+        cls.server_type = tm.get_cas_host_type(cls.s)
+
+    def test_option_type(self):
+        input1 = InputLayer(n_channels = 1, width = 28, height = 28)
+        conv1 = Conv2d(2)(input1)
+        conv2 = Conv2d(2)(input1)
+        concat1 = Concat(src_layers = [conv1, conv2])
+        output1 = OutputLayer(n=2)(concat1)
+        model1 = Model(conn = self.s, inputs = [input1], outputs = output1)
+        model1.compile()
+        model1.print_summary()
+
+        model2 = Model(conn = self.s, inputs = input1, outputs = output1)
+        model2.compile()
+        model2.print_summary()
+        model2.plot_network()
+
+    def test_concat(self):
+        input1 = InputLayer(n_channels = 1, width = 28, height = 28)
+        input2 = InputLayer(n_channels = 3, width = 28, height = 28)
+        conv1 = Conv2d(2)(input1)
+        conv2 = Conv2d(2)(input1)
+        conv3 = Conv2d(2)(input2)
+        output2 = OutputLayer(n=2)(conv3)
+        concat1 = Concat(src_layers = [conv1, conv2, conv3])
+        output1 = OutputLayer(n=2)(concat1)
+        model1 = Model(conn = self.s, inputs = [input1, input2], outputs = [output1, output2])
+        model1.compile()
+        model1.print_summary()
+
+    def test_inputs(self):
+        try:
+            conv1 = Conv2d(2)
+            output1 = OutputLayer(2)(conv1)
+            model1 = Model(conn = self.s, inputs = [conv1], outputs = [output1])
+            model1.compile()
+        except DLPyError:
```

Comment:
If you are expecting an exception here, you can use the self.assertRaises(...) method to test for that.